target/linux/layerscape/modules.mk

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (C) Jiang Yutang <yutang.jiang@nxp.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+I2C_IMX_MODULES:= \
+  CONFIG_I2C_IMX:drivers/i2c/busses/i2c-imx
+
+define KernelPackage/i2c-imx
```

Comment:
You can simplify all this by enabling i2c and sdhci in the kernel config and dropping modules.mk entirely